odc/stac/eo3/_eo3converter.py

```diff
@@ -245,13 +252,14 @@ def _item_to_ds(
     )
     _item = parse_item(item, md)
 
-    return _to_dataset(_item, item.properties, ds_uuid, product)
+    return _to_dataset(_item, item.properties, ds_uuid, product, asset_url)
 
 
 def stac2ds(
     items: Iterable[pystac.item.Item],
     cfg: Optional[ConversionConfig] = None,
     product_cache: Optional[Dict[str, DatasetType]] = None,
+    asset_urls: Optional[Iterable[str]] = None,
```

Comment:
Given that a pystac Item has `assets` with their own hrefs, I think this param might be better called something a little less potentially ambiguous, such as `item_urls`.